src/accounts_refactor.rs

```diff
@@ -0,0 +1,339 @@
+    /// Whether the account is used.
+    pub used: bool,
+    /// Optional ledger scan.
+    scan: Option<BackgroundKeyScan<L>>,
+    /// The list of asset codes of the account.
+    assets: Vec<AssetCode>,
+    /// The list of records of the account.
+    records: Vec<RecordInfo>,
+    /// The table of balances with corresponding asset code.
+    balances: HashMap<AssetCode, U256>,
+    /// The time when the account was created.
+    created_time: DateTime<Local>,
+    /// The last time when the account was modified.
+    modified_time: DateTime<Local>,
+}
+
+impl<L: Ledger, Key: KeyPair> Account<L, Key> {
+    #![allow(dead_code)]
+
+    /// Get the account key.
+    pub fn key(&self) -> &Key {
+        &self.key
+    }
+
+    /// Get the account public key.
+    pub fn pub_key(&self) -> Key::PubKey {
+        self.key.pub_key()
+    }
+
+    /// Get the account description.
+    pub fn description(&self) -> String {
+        self.description.clone()
+    }
+
+    /// Check whether the account is used.
+    pub fn used(&self) -> bool {
+        self.used
+    }
+
+    /// Get the optional ledger scan.
+    pub fn scan(&self) -> Option<BackgroundKeyScan<L>> {
+        self.scan.clone()
+    }
+
+    /// Get the list of asset codes.
+    pub fn assets(&self) -> Vec<AssetCode> {
+        self.assets.clone()
+    }
+
+    /// Get the list of records.
+    pub fn records(&self) -> Vec<RecordInfo> {
+        self.records.clone()
+    }
+
+    /// Get the table of balances.
+    pub fn balances(&self) -> HashMap<AssetCode, U256> {
+        self.balances.clone()
+    }
+
+    /// Get the created time.
+    pub fn created_time(&self) -> DateTime<Local> {
+        self.created_time
+    }
+
+    /// Get the modified time.
+    pub fn modified_time(&self) -> DateTime<Local> {
+        self.modified_time
+    }
+}
+
+pub type AccountsStore<L, Key, PubKey> = KeyValueStore<PubKey, Account<L, Key>>;
+
+/// An editor to create or update the account or accounts store.
+pub struct AccountEditor<'a, L: Ledger, Key: KeyPair + DeserializeOwned + Serialize> {
+    account: Account<L, Key>,
+    store: &'a mut AccountsStore<L, Key, Key::PubKey>,
+}
+
+impl<'a, L: Ledger, Key: KeyPair + DeserializeOwned + Serialize> AccountEditor<'a, L, Key> {
+    /// Create an account editor.
+    fn new(
+        store: &'a mut AccountsStore<L, Key, Key::PubKey>,
+        account: Account<L, Key>,
+    ) -> AccountEditor<'a, L, Key> {
+        Self { account, store }
+    }
+
+    /// Set the account as used.
+    pub fn set_used(mut self) -> Self {
+        self.account.used = true;
+        self.account.modified_time = Local::now();
+        self
+    }
+
+    /// Set the account as unused.
+    pub fn set_unused(mut self) -> Self {
+        self.account.used = false;
+        self.account.modified_time = Local::now();
+        self
+    }
+
+    /// Set the optional ledger scan.
+    pub fn set_scan(mut self, scan: Option<BackgroundKeyScan<L>>) -> Self {
```

Comment:
This should probably be `pub(crate)` since key scans are created and managed internally